lib/request.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const http = require('https');
+
+const post = (url, data, cb) => {
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Content-Length': Buffer.byteLength(data),
+    },
+  };
+
+  let errored = false;
+
+  const req = http.request(url, options, (res) => {
+    res.setEncoding('utf8');
+
+    const { statusCode } = res;
+
+    let body = '';
+    res.on('data', (chunk) => {
+      body += chunk.toString();
+    });
+
+    res.on('end', () => {
+      if (!errored) {
+        cb(null, { statusCode }, body);
```

Comment:
This is brittle, because you'll get mochibake if any character boundaries get split.  I'd recommend doing it this way instead:

```js
const body = []
res.on('data', chunk => body.push(chunk))
res.on('end', () => {
  if (!errored) {
    cb(null, { statusCode }, Buffer.concat(body).toString())
  }
})
```

Reply:
thank you! i read about this years ago and forgot again, now i will remember it as the pattern to use.